sqlalchemy_drill/jdbc.py

```diff
@@ -36,45 +37,41 @@ class DrillDialect_jdbc(DrillDialect):
 
     def __init__(self, *args, **kwargs):
         super(DrillDialect_jdbc, self).__init__(*args, **kwargs)
-        self.jdbc_driver_path = os.environ.get('DRILL_JDBC_DRIVER_PATH')
-        self.jdbc_jar_name = os.environ.get('DRILL_JDBC_JAR_NAME')
 
-        if self.jdbc_driver_path is None:
-            raise Exception('To connect to Drill via JDBC, you must set the DRILL_JDBC_DRIVER path to the location of the Drill JDBC driver.')
-
-        if self.jdbc_jar_name is None:
-            raise Exception(
-                'To connect to Drill via JDBC, you must set the DRILL_JDBC_JAR_NAME environment variable.')
+        # The user is responsible for starting the JVM with the class path, but we can
+        # do some sanity checks to make sure they are pointed in the right direction.
+        if not jpype.isJVMStarted():
+            raise Exception("The JVM must be started before connecting to a JDBC driver.")
+        try:
+            jpype.JClass("org.apache.drill.jdbc.Driver")
+        except TypeError:
+            err = "The drill JDBC driver class was not located in the CLASSPATH `%s`"%str(jpype.java.lang.System.getProperty('java.class.path'))
+            raise Exception(err)
```

Comment:
Here is the big difference in that jpype.dbapi2 needs to be started outside of the sql so that it started once and from the proper thread.  If started from another thread it may deadlock entirely.